core/src/main/java/io/spine/core/Events.java

```diff
@@ -218,6 +219,52 @@ public static boolean isExternal(EventContext context) {
     }
 
     /**
+     * Obtains the compacted version of the event, which should be used for storing.
+     *
+     * <p>A compacted version doesn't contain:
+     * <ul>
+     *     <li>the enrichment from the event context;</li>
+     *     <li>the enrichment from the origin;</li>
+     *     <li>nested origins if the origin is {@link EventContext}.</li>
+     * </ul>
+     *
+     * @param event the event to compact
+     * @return the compacted event
+     */
+    @Internal
+    public static Event compact(Event event) {
+        final EventContext context = event.getContext();
+        final EventContext.Builder resultContext = context.toBuilder()
+                                                          .clearEnrichment();
+        final EventContext.OriginCase originCase = resultContext.getOriginCase();
+        switch (originCase) {
+            case EVENT_CONTEXT:
+                resultContext.setEventContext(context.getEventContext()
+                                                     .toBuilder()
+                                                     .clearOrigin()
```

Comment:
As discussed, do not clear this. And please mention that in docs.

Reply:
Done.